### PR TITLE
Created first communication interfaces

### PR DESCRIPTION
**Pull request by CremaLuca**
I introduced the library to the generic communication interfaces, independent from the SMS communication channels.
Let me know if your interfaces differ even slightly so that we can work on merging correctly all the groups.

SMSHandler is just a test class to have on the right package, leave it as it is for now, it's not important.

### CODE REVIEW

File: .idea/modules.xml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <project version="4">
   <component name="ProjectModuleManager">
     <modules>
-      <module fileurl="file://$PROJECT_DIR$/smslibrary.iml" filepath="$PROJECT_DIR$/smslibrary.iml" />
+      <module fileurl="file://$PROJECT_DIR$/SMSLibProject.iml" filepath="$PROJECT_DIR$/SMSLibProject.iml" />
```
Comment (Cogno-IDU) [inline .idea/modules.xml]: I have no idea what this does but let's leave android files to android studio

Comment (CremaLuca) [reply .idea/modules.xml]: I have no idea either ¯\_(ツ)_/¯ Android studio did this by himself so I hope it knows what's it's doing.

File: smslibrary/src/main/java/com/eis/communication/Message.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+    /**
+     * Retrieves the data sent or to be sent in the network
+     * @return data contained in this message
+     */
+    D getData();
+
+    /**
+     * Retrieves the sender or the destination
+     * @return the peer this message is for or is from
```
Comment (Cogno-IDU) [inline smslibrary/src/main/java/com/eis/communication/Message.java]: You can just be more generic here, like `Peer associated with this message`

File: smslibrary/src/main/java/com/eis/communication/Peer.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+/**
+ * Represents a user in the network
+ * @param <A> The type of address used by the network
+ */
+public interface Peer<A> extends Serializable {
+
+    A getAddress();
+
+    void setAddress(A address);
```
Comment (Cogno-IDU) [inline smslibrary/src/main/java/com/eis/communication/Peer.java]: Missing specifications for this methods (even though the name makes it obvious)

File: smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+import com.eis.communication.CommunicationHandler;
+import com.eis.communication.Message;
+
+
+public class SMSHandler implements CommunicationHandler {
+
+    @Override
+    public void sendMessage(Message message) {
+        //Test pull request
```
Comment (Cogno-IDU) [inline smslibrary/src/main/java/com/eis/smslibrary/SMSHandler.java]: Don't worry, the pull request works fine 😉

File: smslibrary/src/main/java/com/eis/communication/Peer.java (hunk 3 above)
Comment (CremaLuca) [reply smslibrary/src/main/java/com/eis/communication/Peer.java]: Holy cow! I forgot to add them, thanks.

**Comment by CremaLuca** [comment]
Applied the suggested changes. I think we can proceed with merge as this is the most basic interface we can think of and doesn't need further approbation.
We could wait for someone from g1 or g3 to approve this anyway if you wish.

**Comment by Cogno-IDU** [comment]
> Applied the suggested changes. I think we can proceed with merge as this is the most basic interface we can think of and doesn't need further approvation.

Yeah, gonna merge this, we don't have much time to do everything, better move onto something else 👍

**Comment by Cogno-IDU** [comment]
ops, wrong button

**Comment by Cogno-IDU** [comment]
Alright, now it's merged

**Pull request merged**